### ISSUE TITLE
setup mock website for stakeholder feedback

### ISSUE THREAD

**Issue by bbest** [mhk-env/mhk-env.github.io#2]

**Comment by bbest** [comment]
Per commit https://github.com/mhk-env/bakerydemo/commit/6d06386e6732c5e3a056f6a9748ee6c66976af8b, I added the following web services to [mhk-env.us](http://mhk-env.us):

- **[cms](http://cms.mhk-env.us)**.mhk-env.us: content management system using wagtail, initially populated with [bakerydemo](https://github.com/wagtail/bakerydemo)
  ![image](https://user-images.githubusercontent.com/2837257/61683288-dc8ad200-acc9-11e9-92f4-a12d2ef17e3c.png)

- **[rstudio](http://rstudio.mhk-env.us)**.mhk-env.us: RStudio IDE for R
  ![image](https://user-images.githubusercontent.com/2837257/61684133-2de89080-accd-11e9-82d2-fc531b43272a.png)

- **[shiny](http://rstudio.mhk-env.us)**.mhk-env.us: Shiny apps
  ![image](https://user-images.githubusercontent.com/2837257/61686286-047f3300-acd4-11e9-9c6f-060d62b2a452.png)

**Comment by bbest** [comment]
Grabbed `nginx.tmpl` from [[FEATURE] Allow multiple ports per virtual host by frederikbosch · Pull Request #1157 · jwilder/nginx-proxy](https://github.com/jwilder/nginx-proxy/pull/1157) to get goin with these two small commits:

* [nginx-proxy w/ multiple ports for shiny:3838 rstudio:8787 fixes https… · mhk-env/bakerydemo@5ec6d84](https://github.com/mhk-env/bakerydemo/commit/5ec6d84460179f96bcb6a5b2bf04bb35637670fc)
* [init +nginx-proxy +rstudio-shiny · mhk-env/bakerydemo@6d06386](https://github.com/mhk-env/bakerydemo/commit/6d06386e6732c5e3a056f6a9748ee6c66976af8b)

**Comment by bbest** [comment]
## 2019-07-22 wagtail initial setup

See: [mhk-env/bakerydemo: `docker-compose.yml`](https://github.com/mhk-env/bakerydemo/blame/master/docker-compose.yml)

### login to server

- [Installing SSHPass](https://gist.github.com/arunoda/7790979)
- [pipe - piping a password located in a file to the ssh command - Unix & Linux Stack Exchange](https://unix.stackexchange.com/questions/223491/piping-a-password-located-in-a-file-to-the-ssh-command)

```
sshpass -f ~/.mhk-env.us_bbest.password ssh bbest@mhk-env.us
```

### docker run wagtail

[Your first Wagtail site — Wagtail 2.5.1 documentation](http://docs.wagtail.io/en/v2.5.1/getting_started/tutorial.html)

Forked:
- [wagtail/bakerydemo](https://github.com/wagtail/bakerydemo)

```bash
#sudo apt install python3-pip
sudo apt install docker-compose

git clone https://github.com/mhk-env/bakerydemo.git wagtail
cd wagtail

#docker network create netproxy
# rm wag*: docker ps -aq --filter name=wag* | xargs docker rm
# rm *: docker ps -aq | xargs docker rm
docker-compose up --build -d
# wait until all containers are up
docker-compose run app /venv/bin/python manage.py load_initial_data
```

```bash
# stop
docker-compose stop
# start
docker-compose up -d
# list
docker ps -a
```

### edit docker-compose.yml

```bash
cd wagtail
vi docker-compose.yml
```

```
app:
  networks:
    - netproxy
  environment:
    VIRTUAL_HOST: mhk-env.us
  expose:
    - "8000"
networks:
  netproxy:
    external:
      name: netproxy
```

### setup reverse proxy

- [jwilder/nginx-proxy: Automated nginx proxy for Docker containers using docker-gen](https://github.com/jwilder/nginx-proxy)

- [Automated Nginx Reverse Proxy for Docker ·](http://jasonwilder.com/blog/2014/03/25/automated-nginx-reverse-proxy-for-docker/)

#### Old

```bash
docker network create --driver bridge netproxy

docker run -d -ti --rm -p 80:80 \
  -v /var/run/docker.sock:/tmp/docker.sock:ro \
  --network netproxy \
  --name proxy jwilder/nginx-proxy
```

#### New

Fold into `wagtail/docker-compose.yml`

Need same rstudio-shiny container to serve two different ports:

```
upstream shiny.mhk-env.us {
	server 172.21.0.3:3838;
}

upstream rstudio.mhk-env.us {
	server 172.21.0.3:8787;
}
```

**SOLUTION**: [[FEATURE] Allow multiple ports per virtual host by frederikbosch · Pull Request #1157 · jwilder/nginx-proxy](https://github.com/jwilder/nginx-proxy/pull/1157)

- used [`nginx.tmpl`](https://raw.githubusercontent.com/jwilder/nginx-proxy/6f292f7630d9e0d663b6ab0ce2f318e89e5c6bb2/nginx.tmpl)

#### messing with nginx-proxy

```
docker logs proxy

# extract the generated nginx configuration
docker exec wagtail_nginx-proxy_1 cat /etc/nginx/conf.d/default.conf

# terminal
docker exec -it wagtail_nginx-proxy_1 bash
cat /etc/nginx/conf.d/default.conf

# manually reload nginx
docker exec wagtail_nginx-proxy_1 nginx -s reload

# edit conf
apt-get update
apt-get install vim
vi /etc/nginx/conf.d/default.conf
```
